File: agentwallet/packages/sdk-python/src/agentwallet/resources/analytics.py
"""Analytics sub-resource."""

from __future__ import annotations

from typing import TYPE_CHECKING

from ..types import AnalyticsSummary

if TYPE_CHECKING:
    from ..client import AgentWallet


class AnalyticsResource:
    def __init__(self, client: AgentWallet):
        self._client = client

    async def summary(self, days: int = 30) -> AnalyticsSummary:
        data = await self._client.get("/analytics/summary", params={"days": days})
        return AnalyticsSummary(**data)

    async def daily(
        self, days: int = 30, agent_id: str | None = None
    ) -> list[dict]:
        params = {"days": days}
        if agent_id:
            params["agent_id"] = agent_id
        return await self._client.get("/analytics/daily", params=params)

    async def by_agent(self, days: int = 30) -> list[dict]:
        return await self._client.get("/analytics/agents", params={"days": days})

File: agentwallet/packages/sdk-ts/src/resources/analytics.ts
import type { AgentWallet } from '../client';
import type { AnalyticsSummary } from '../types';

export class AnalyticsResource {
  constructor(private client: AgentWallet) {}

  /** Get analytics summary */
  async summary(days = 30): Promise<AnalyticsSummary> {
    return this.client.get<AnalyticsSummary>('/analytics/summary', { days });
  }

  /** Get daily analytics breakdown */
  async daily(options?: { days?: number; agent_id?: string }): Promise<Record<string, unknown>[]> {
    return this.client.get<Record<string, unknown>[]>('/analytics/daily', {
      days: options?.days ?? 30,
      agent_id: options?.agent_id,
    });
  }

  /** Get analytics grouped by agent */
  async byAgent(days = 30): Promise<Record<string, unknown>[]> {
    return this.client.get<Record<string, unknown>[]>('/analytics/agents', { days });
  }
}
